src/tlo/population.py

```diff
@@ -11,8 +10,31 @@
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
-PatientDetails: TypeAlias = Tuple[Any]
-PatientDetailsType: TypeAlias = Type[Tuple[Any]]
+
+class PatientDetails:
+    """Read-only memoized view of population dataframe row."""
+    
+    def __init__(self, population_dataframe: pd.DataFrame, person_id: int):
+        self._population_dataframe = population_dataframe
+        self._person_id = person_id
+        self._property_cache: Dict[str, Any] = {}
+        
+    def __getitem__(self, key: str) -> Any:
+        cached_value = self._property_cache.get(key)
+        if cached_value is not None:
```

Comment:
I'm sure it's fine for now, since as far as I've seen the convention for missing data in the simulation is NaN or Inf; but I can forsee a situation where the DataFrame value genuinely is `None`. In which case with this code we'd be reading the property over and over again from the DataFrame.

But I also don't know what else we can do in this situation anyway.

Reply:
Yeah that's a good point, definitely worth making sure this is future proofed. We could great a dedicated sentinel type to use here or index directly and wrap in a `try ... except` block.